library/src/scala/tasty/reflect/Kernel.scala

```diff
@@ -865,6 +865,14 @@ trait Kernel {
   /** Is this type an instance of a non-bottom subclass of the given class `cls`? */
   def Type_derivesFrom(self: Type)(cls: ClassDefSymbol)(implicit ctx: Context): Boolean
 
+  /** Is this type a function type?
+   *
+   *  @return true if the dealised type of `self` without refinement is `FunctionN[T1, T2, ..., Tn]`
+   *
+   *  @note `List[Int]` is not a function type, despite that `List[Int] <:< Int => Int`.
+   */
+  def Type_isFunctionType(self: Type)(implicit ctx: Context): Boolean
+
```

Comment:
The specification of the method may raise some concerns. However, in the setting of meta-programming, this is can be justified, as it's well-known that reflection breaks referential transparency. E.g. `1 + 1 == 2`, but `'{1 + 1} != '2`. For this method, the semantic reasoning breaks at the type level -- an intended effect that's useful in meta-programming.